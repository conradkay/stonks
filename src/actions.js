export const selectStonkInInventory = (index) => ({
    type: 'SELECT_STONK_IN_INVENTORY',
    index: index
});
export const setShopValue = (index, amount) => ({
    type: 'SET_VALUE_IN_SHOP', index, number: amount
});
export const sellStonks = () => ({
    type: 'SELL_STONKS'
});
export const unSelectAllInventory = () => ({
    type: 'UNSELECT_ALL_INVENTORY'
});
export const selectAllInventory = () => ({
    type: 'SELECT_ALL_INVENTORY'
});
export const addToCart = (stonk, amount) => {
    const largestId = Math.floor((Math.random() * 10000000));
    return {
        type: 'ADD_TO_CART',
        stonk,
        id: largestId,
        amount
    };
};
export const changeStonkAmountInInventory = (amount, index) => ({
    type: 'CHANGE_STONK_AMOUNT_IN_INVENTORY', index, amount
});
export const changeAmount = (amount, index) => ({
    type: 'CHANGE_AMOUNT', amount, index
});
export const removeStonk = (index) => ({ type: 'REMOVE_STONK', index: index });
export const selectAllCart = () => ({ type: 'SELECT_ALL_CART' });
export const unSelectAllCart = () => ({ type: 'UNSELECT_ALL_CART' });
export const buyStonks = () => ({ type: 'BUY_STONKS' });
export const selectStonkInCart = (index) => ({
    type: 'SELECT_STONK_IN_CART',
    index: index
});
export const changeFilter = (newFilter) => ({ type: 'CHANGE_FILTERS', newFilter });
export const openSnackbar = (message, variant) => ({
    type: 'OPEN_SNACKBAR',
    message,
    variant
});
export const closeSnackbar = () => ({ type: 'CLOSE_SNACKBAR' });
export const sellStonk = () => ({ type: 'SELL_STONKS' });
//# sourceMappingURL=actions.js.map