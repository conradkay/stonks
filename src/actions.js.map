{"version":3,"file":"actions.js","sourceRoot":"","sources":["actions.ts"],"names":[],"mappings":"AACA,MAAM,CAAC,MAAM,sBAAsB,GAAG,CAAC,KAAa,EAAU,EAAE,CAAC,CAAC;IAChE,IAAI,EAAE,2BAA2B;IACjC,KAAK,EAAE,KAAK;CACb,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,YAAY,GAAG,CAAC,KAAa,EAAE,SAAiB,EAAU,EAAE,CAAC,CAAC;IACzE,IAAI,EAAE,mBAAmB,EAAE,KAAK,EAAE,MAAM,EAAE,SAAS;CACpD,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,UAAU,GAAG,GAAW,EAAE,CAAC,CAAC;IACvC,IAAI,EAAE,aAAa;CACpB,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,oBAAoB,GAAG,GAAW,EAAE,CAAC,CAAC;IACjD,IAAI,EAAE,wBAAwB;CAC/B,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,kBAAkB,GAAG,GAAW,EAAE,CAAC,CAAC;IAC/C,IAAI,EAAE,sBAAsB;CAC7B,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,SAAS,GAAG,CAAC,KAAY,EAAE,MAAc,EAAU,EAAE;IAChE,MAAM,SAAS,GAAW,IAAI,CAAC,KAAK,CAAC,CAAC,IAAI,CAAC,MAAM,EAAE,GAAG,QAAQ,CAAC,CAAC,CAAA;IAChE,OAAO;QACL,IAAI,EAAE,aAAa;QACnB,KAAK;QACL,EAAE,EAAE,SAAS;QACb,MAAM;KACP,CAAA;AACH,CAAC,CAAA;AACD,MAAM,CAAC,MAAM,4BAA4B,GAAG,CAAC,MAAc,EAAE,KAAa,EAAE,EAAE,CAAC,CAAC;IAC9E,IAAI,EAAE,kCAAkC,EAAE,KAAK,EAAE,MAAM;CACxD,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,YAAY,GAAG,CAAC,MAAc,EAAE,KAAa,EAAU,EAAE,CAAC,CAAC;IACtE,IAAI,EAAE,eAAe,EAAE,MAAM,EAAE,KAAK;CACrC,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,WAAW,GAAG,CAAC,KAAa,EAAU,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,cAAc,EAAE,KAAK,EAAE,KAAK,EAAE,CAAC,CAAA;AAC9F,MAAM,CAAC,MAAM,aAAa,GAAG,GAAW,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,iBAAiB,EAAE,CAAC,CAAA;AACxE,MAAM,CAAC,MAAM,eAAe,GAAG,GAAW,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,mBAAmB,EAAE,CAAC,CAAA;AAE5E,MAAM,CAAC,MAAM,SAAS,GAAG,GAAW,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,YAAY,EAAE,CAAC,CAAA;AAC/D,MAAM,CAAC,MAAM,iBAAiB,GAAG,CAAC,KAAa,EAAU,EAAE,CAAC,CAAC;IAC3D,IAAI,EAAE,sBAAsB;IAC5B,KAAK,EAAE,KAAK;CACb,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,YAAY,GAAG,CAAC,SAAiB,EAAU,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,gBAAgB,EAAE,SAAS,EAAE,CAAC,CAAA;AAClG,MAAM,CAAC,MAAM,YAAY,GAAG,CAAC,OAAe,EAAE,OAAgB,EAAU,EAAE,CAAC,CAAC;IAC1E,IAAI,EAAE,eAAe;IACrB,OAAO;IACP,OAAO;CACR,CAAC,CAAA;AACF,MAAM,CAAC,MAAM,aAAa,GAAG,GAAW,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,gBAAgB,EAAE,CAAC,CAAA;AACvE,MAAM,CAAC,MAAM,SAAS,GAAG,GAAW,EAAE,CAAC,CAAC,EAAE,IAAI,EAAE,aAAa,EAAE,CAAC,CAAA"}