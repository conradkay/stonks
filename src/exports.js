export { stonks } from './stonks';
export { Inventory } from './inventory/inventory';
export { addToCart, changeFilter, openSnackbar, closeSnackbar, selectStonkInCart, unSelectAllCart, removeStonk, selectAllCart, buyStonks, sellStonk, changeAmount, selectStonkInInventory, unSelectAllInventory, selectAllInventory, sellStonks, changeStonkAmountInInventory, setShopValue } from './actions';
export { Home } from './home/home';
export { categories, getClassSnackbarVariant, defaultState, itemBetweenIndexes } from './utils';
export { withRoot } from './withroot';
export { SnackbarWrap } from './allPage/snackbarWrap';
export { Shop } from './shop/shop';
export { Cart } from './cart/cart';
export { Footer } from './allPage/footer';
export { Header } from './allPage/header';
export { Search } from './allPage/search';
export { reducer } from './reducers/root';
//# sourceMappingURL=exports.js.map