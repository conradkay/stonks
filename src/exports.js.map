{"version":3,"file":"exports.js","sourceRoot":"","sources":["exports.ts"],"names":[],"mappings":"AACA,OAAO,EAAE,MAAM,EAAE,MAAM,UAAU,CAAA;AACjC,OAAO,EAAE,SAAS,EAAE,MAAM,uBAAuB,CAAA;AACjD,OAAO,EACL,SAAS,EAAE,YAAY,EAAE,YAAY,EAAE,aAAa,EAAE,iBAAiB,EAAE,eAAe,EACxF,WAAW,EAAE,aAAa,EAAE,SAAS,EAAE,SAAS,EAAE,YAAY,EAAE,sBAAsB,EACtF,oBAAoB,EAAE,kBAAkB,EAAE,UAAU,EAAE,4BAA4B,EAAE,YAAY,EACjG,MAAM,WAAW,CAAA;AAClB,OAAO,EAAE,IAAI,EAAE,MAAM,aAAa,CAAA;AAClC,OAAO,EAAE,UAAU,EAAE,uBAAuB,EAAE,YAAY,EAAE,kBAAkB,EAAE,MAAM,SAAS,CAAA;AAC/F,OAAO,EAAE,QAAQ,EAAE,MAAM,YAAY,CAAA;AACrC,OAAO,EAAE,YAAY,EAAE,MAAM,wBAAwB,CAAA;AACrD,OAAO,EAAE,IAAI,EAAE,MAAM,aAAa,CAAA;AAClC,OAAO,EAAE,IAAI,EAAE,MAAM,aAAa,CAAA;AAClC,OAAO,EAAE,MAAM,EAAE,MAAM,kBAAkB,CAAA;AACzC,OAAO,EAAE,MAAM,EAAE,MAAM,kBAAkB,CAAA;AACzC,OAAO,EAAE,MAAM,EAAE,MAAM,kBAAkB,CAAA;AACzC,OAAO,EAAE,OAAO,EAAE,MAAM,iBAAiB,CAAA"}