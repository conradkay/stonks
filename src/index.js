import React from 'react';
import { render } from 'react-dom';
import './index.css';
import { Contact } from './contact/contact';
import { composeWithDevTools } from 'redux-devtools-extension';
import { Provider } from 'react-redux';
import { Home, Cart, Shop, defaultState, reducer, Inventory } from './exports';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { createStore } from 'redux';
const store = createStore(reducer, defaultState, composeWithDevTools());
const Wrapper = () => {
    return (React.createElement(Provider, { store: store },
        React.createElement(Router, { basename: "/stonk" },
            React.createElement(Switch, null,
                React.createElement(Route, { exact: true, path: '/', render: () => React.createElement(Home, null) }),
                React.createElement(Route, { path: '/cart/', render: () => React.createElement(Cart, null) }),
                React.createElement(Route, { path: '/contact/', render: () => React.createElement(Contact, null) }),
                React.createElement(Route, { path: '/shop/', render: () => React.createElement(Shop, null) }),
                React.createElement(Route, { path: '/inventory/', render: () => React.createElement(Inventory, null) })))));
};
render(React.createElement(Wrapper, null), document.getElementById('root'));
//# sourceMappingURL=index.js.map