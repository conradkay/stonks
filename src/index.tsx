import React from 'react'
import { render } from 'react-dom'
import './index.css'
import { Contact } from './contact/contact'
import { composeWithDevTools } from 'redux-devtools-extension'
import { Provider } from 'react-redux'
import Inventory from './inventory/inventory'
import { Home, Cart, Shop, defaultState, reducer, SnackbarWrap } from './exports'
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom'
import { createStore } from 'redux'

// let persistedState: State = defaultState
// if(localStorage.getItem('reduxState')) {
//   persistedState = JSON.parse(localStorage.getItem('reduxState') || '')
// }

const store = createStore(reducer, defaultState, composeWithDevTools())
// store.subscribe(() => {
//   localStorage.setItem('reduxState', JSON.stringify(store.getState()))
// })

const Wrapper = () => {
  return (
  <Provider store={store}>
    <div>
      <SnackbarWrap />
      <Router basename="/stonk">
        <Switch>
          <Route exact={true} path={'/'} render={() => <Home />}/>
          <Route path={'/cart/'} render={() => <Cart />}/>
          <Route path={'/contact/'} render={() => <Contact />}/>
          <Route path={'/shop/'} render={() => <Shop />}/>
          <Route path={'/inventory/'} render={() => <Inventory />}/>
        </Switch>
      </Router>
    </div>
  </Provider>
  )
}
render(
  <Wrapper />,
  document.getElementById('root')
)
