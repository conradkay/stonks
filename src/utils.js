import { stonks } from './exports';
export const categories = [
    'food', 'realEstate', 'office', 'service'
];
export const itemBetweenIndexes = (parent, index, newItem) => {
    return [...parent.slice(0, index), newItem, ...parent.slice(index + 1)];
};
export const getClassSnackbarVariant = (variant, classes) => {
    if (variant === 'success') {
        return classes.success;
    }
    else if (variant === 'warning') {
        return classes.warning;
    }
    else if (variant === 'error') {
        return classes.error;
    }
    else {
        return classes.error;
    }
};
export const defaultState = {
    amountStonksShop: [
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1
    ],
    allSelected: false,
    cart: stonks,
    inventory: [...stonks],
    activeFilters: [],
    money: 1000,
    snackbar: { open: false, message: 'dont show', variant: 'success' },
};
//# sourceMappingURL=utils.js.map