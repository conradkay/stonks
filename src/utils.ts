import { State, Variant, stonks, Stonk } from './exports'
/* used in header component as such and
*  <Link style={getStylesHeader(currComp === 'contact')} />
*/
export const getStylesHeader = (conditionTrue: boolean): { textDecoration: 'underline' } | {} => {
  if (conditionTrue) {
    return { textDecoration: 'underline' }
  }
  else {
    return {}
  }
}
// used as parent: [...itemBetweenIndexes(state.parent, 3, {})]
export const itemBetweenIndexes = (parent: Array<any>, index: number, newItem: any) => {
  return [...parent.slice(0, index), newItem, ...parent.slice(index + 1)]
}

export const getClassSnackbarVariant = (variant: Variant, classes: { success: string; warning: string; error: string }) => {
  // function to bypass rules on typescript [] notation, used in snackbarWrap component
  if (variant === 'success') {
    return classes.success
  }
  else if (variant === 'warning') {
    return classes.warning
  }
  else if (variant === 'error') {
    return classes.error
  }
  else {
    return classes.error
  }
}
export const generateOnes = (amount: number): number[] => { // generates some ones to put into state.shopAmounts
  const arr = [1]
  for (let i = 0; i < amount - 1; i++) {
    arr.push(1)
  }
  return arr
}
export const defaultState: State = {
  amountStonksShop: generateOnes(25),
  cart: [
    ...stonks.slice(0, -1),
    { ...stonks[stonks.length - 1], selected: true }
  ],
  inventory: [
    ...stonks.slice(0, -1),
    { ...stonks[stonks.length - 1], selected: true }
  ],
  activeFilters: [],
  money: 1000,
  snackbar: { open: false, message: 'dont show', variant: 'success' },
}
export const exampleStonk: Stonk = {
  amount: 2,
  id: 1,
  description: `floopderop`,
  name: 'floop',
  price: 101,
  pathToImage: 'https://',
  sellAmount: 1,
  selected: false
}
