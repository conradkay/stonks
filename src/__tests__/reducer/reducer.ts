// hello mate, this is the dev branch
import { Stonk, reducer } from 'src/exports'
import { defaultState, exampleStonk } from 'src/utils'
import * as actions from 'src/actions'
import { addToCart } from '../../reducers/addToCart'
import { createStore } from 'redux'
const store = createStore(reducer, defaultState)
beforeEach(() => {
  store.dispatch({ type: 'RESET' })
})
describe('reducer', () => {
  it('should add stonks to cart', () => {
    store.dispatch({ type: 'SET_STATE_TEST', newState: { ...defaultState, cart: [] } })
    store.dispatch(actions.addToCart(exampleStonk, 2)) // case 3, new stonk in empty cart
    expect(store.getState().cart).toEqual([{ ...exampleStonk, amount: 2, id: store.getState().cart[0].id }])
    store.dispatch({ type: 'RESET' }) // reset for other tests in it
    store.dispatch(actions.addToCart(exampleStonk, 9)) // case 1, new stonk
    store.dispatch(actions.addToCart(exampleStonk, 1)) // expect adding existing stonk to add to amount
    expect(store.getState()).toMatchSnapshot()
    const stonk = {
      ...exampleStonk,
      id: store.getState().cart[store.getState().cart.length - 1].id,
      amount: 12
    }
    const expected = [
      ...defaultState.inventory,
      stonk
    ]
    expect(store.getState().cart).toEqual(expected)
    expect(addToCart({ ...defaultState, cart: [] }, { ...actions.addToCart(stonk, 11), id: 1 }).cart).toEqual([{ ...stonk, id: 1 }])
  })
  it('should buy stonks', () => {

  })
  it('should sell stonks', () => {

  })
  it('should change amount in shop/cart/inventory', () => {

  })
  it('should select stonk in shop/cart/inventory', () => {

  })
  it('should select all in shop/cart/inventory', () => {

  })
  it('should unselect all in cart/inventory', () => {
    const expectedCart = store.getState().cart.map((stonk: Stonk) => {
      return { ...stonk, selected: false }
    })
    const expectedInventory = store.getState().inventory.map((stonk: Stonk) => {
      return { ...stonk, selected: false }
    })
    const expectedState = {
      ...defaultState,
      cart: expectedCart,
      inventory: expectedInventory
    }
    expect(store.getState()).not.toEqual(expectedState)
    store.dispatch(actions.unSelectAllCart())
    store.dispatch(actions.unSelectAllInventory())
    expect(store.getState()).toEqual(expectedState)
  })
  it('should open/close snackbar', () => {
    store.dispatch(actions.openSnackbar('hello', 'error'))
    expect(store.getState().snackbar).toEqual({ open: true, message: 'hello', variant: 'error' })
    store.dispatch(actions.closeSnackbar())
    expect(store.getState().snackbar).toEqual({ open: false, message: 'hello', variant: 'error' })
  })
  it('should return state when action is invalid', () => {
    store.dispatch({ type: 'floop' })
    expect(store.getState()).toEqual(defaultState)
  })
})
