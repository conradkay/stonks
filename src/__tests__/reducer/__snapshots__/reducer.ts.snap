// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer should add stonks to cart 1`] = `
Object {
  "activeFilters": Array [],
  "amountStonksShop": Array [
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
    1,
  ],
  "cart": Array [
    Object {
      "amount": 1,
      "description": "If you have unlocked the freeRealEstateStonk, you are a absolute madman,
  go do die for our sakess man. Seriously dude, how do you do this, we dont even know it is likes
  when Fil fixes a bote with just tape wow man.",
      "id": 1,
      "name": "FREE REAL ESTATE",
      "pathToImage": "https://www.outdoorpainter.com/wp-content/uploads/2017/12/1.-SNOW-Handrahan.-Cold-Morning-Warm-Light-12x16.jpg",
      "price": 1015,
      "selected": false,
      "sellAmount": 1,
    },
    Object {
      "amount": 1,
      "description": "A free government stonk granted to nice guys; the whiteboard stonk is trashes,
   but a nice start, you could draw on your hand, with squidwards special secretions instead,
   or with a lightbulb with the lemon sauce to make secret codes, honestly the whiteboard was a mistake
   when the old people die so will the stonk.",
      "id": 0,
      "name": "whiteboard",
      "pathToImage": "http://cdn.ebaumsworld.com/2015/05/14/043327/84584305/tumblr_mm16sq2gsb1rziftho1_1280.jpg",
      "price": 13,
      "selected": false,
      "sellAmount": 1,
    },
    Object {
      "amount": 1,
      "description": "The cheese stonk is like the best stonk, i mean milk is the same but the italian people cant eat it,
   3 million italianos * 10 cheeses a day = 30 million cheeses a day, wowchee. and also fat people eat lots of cheese to help with their water consumption.",
      "id": 2,
      "name": "cheese stonk",
      "pathToImage": "https://skinnyms.com/wp-content/uploads/2012/03/Tuna-Casserole-1.jpg",
      "price": 115,
      "selected": true,
      "sellAmount": 1,
    },
    Object {
      "amount": 12,
      "description": "floopderop",
      "id": 1,
      "name": "floop",
      "pathToImage": "https://",
      "price": 101,
      "selected": false,
      "sellAmount": 1,
    },
  ],
  "inventory": Array [
    Object {
      "amount": 1,
      "description": "If you have unlocked the freeRealEstateStonk, you are a absolute madman,
  go do die for our sakess man. Seriously dude, how do you do this, we dont even know it is likes
  when Fil fixes a bote with just tape wow man.",
      "id": 1,
      "name": "FREE REAL ESTATE",
      "pathToImage": "https://www.outdoorpainter.com/wp-content/uploads/2017/12/1.-SNOW-Handrahan.-Cold-Morning-Warm-Light-12x16.jpg",
      "price": 1015,
      "selected": false,
      "sellAmount": 1,
    },
    Object {
      "amount": 1,
      "description": "A free government stonk granted to nice guys; the whiteboard stonk is trashes,
   but a nice start, you could draw on your hand, with squidwards special secretions instead,
   or with a lightbulb with the lemon sauce to make secret codes, honestly the whiteboard was a mistake
   when the old people die so will the stonk.",
      "id": 0,
      "name": "whiteboard",
      "pathToImage": "http://cdn.ebaumsworld.com/2015/05/14/043327/84584305/tumblr_mm16sq2gsb1rziftho1_1280.jpg",
      "price": 13,
      "selected": false,
      "sellAmount": 1,
    },
    Object {
      "amount": 1,
      "description": "The cheese stonk is like the best stonk, i mean milk is the same but the italian people cant eat it,
   3 million italianos * 10 cheeses a day = 30 million cheeses a day, wowchee. and also fat people eat lots of cheese to help with their water consumption.",
      "id": 2,
      "name": "cheese stonk",
      "pathToImage": "https://skinnyms.com/wp-content/uploads/2012/03/Tuna-Casserole-1.jpg",
      "price": 115,
      "selected": true,
      "sellAmount": 1,
    },
  ],
  "money": 1000,
  "snackbar": Object {
    "message": "dont show",
    "open": false,
    "variant": "success",
  },
}
`;
